server/src/main/java/org/elasticsearch/action/get/TransportGetFromTranslogAction.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.get;
+
+import org.apache.lucene.store.AlreadyClosedException;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.action.support.TransportActions;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.index.engine.Engine;
+import org.elasticsearch.index.mapper.IdFieldMapper;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class TransportGetFromTranslogAction extends HandledTransportAction<MultiGetShardRequest, TransportGetFromTranslogAction.Response> {
```

Comment:
@henningandersen one question about the approach in this old MGet PR that I have. MGet is essentially implemented here by extending `TransportGetFromTranslogAction` to work with a `MultiGetShardRequest` rather than a `GetRequest`. Is that ok with you or do you think there is a better way of doing this? I'm not asking for a full review here, but I want to know that in case I have time, I can bring this PR up to date and count only on fine tuning it later.

Reply:
Yeah, that does not sound too bad.

Reply:
🙂 I mean... I'm open for suggestions!

Reply:
I's a bit hard to see the diff to existing since it is so old. I think we'd want to rename the action too. We could have two actions instead, possibly sharing most of the code. It is not clear to me how complicated that will be? I think that would be my initial gut check, but I would be ok with the approach here too.

Reply:
Ok. Thanks for the suggestion.